src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/formats/html/HtmlDocletWriter.java

```diff
@@ -1144,6 +1144,133 @@ public Content seeTagToContent(Element element, DocTree see, TagletWriterImpl.Co
         }
     }
 
+    // TODO: this method and seeTagToContent share much of the code; consider factoring common pieces out
```

Comment:
Yes. the existing `seeTagToContent` handles all flavors of `@see` tag, and `{@link}`.

It ought to be possible to redirect the use of `@see <reference> <description>` and `{@link}` to this method some some edited variant thereof.

Reply:
> the existing seeTagToContent handles all flavors of @see tag, and {@link}.

All but fake ones. The `seeTagToContent` method uses the path (`DocTreePath`) of the `see` tag being passed. In the case of snippet markup, there's no such tag; so there's noting to derive the path from. Even if we synthesized such a tag, we would need to carefully embed it into the doc comment tree for the duration of the call to `seeTagToContent`, which is way too complicated and might have unforeseen effects.

I would appreciate it, if you could help me refactor this code.